server/session.go

```diff
@@ -182,25 +173,24 @@ func (s *Session) unsubAll() {
 	}
 }
 
-func (s *Session) getRemoteSub(topic string) *RemoteSubscription {
-	s.remoteSubsLock.RLock()
-	defer s.remoteSubsLock.RUnlock()
-
-	return s.remoteSubs[topic]
+// Represents a proxied (remote) session.
+type remoteSession struct {
+	// User id of the proxied session.
+	uid types.Uid
+	// Whether the proxied session is background.
+	isBackground bool
 }
 
-func (s *Session) addRemoteSub(topic string, remoteSub *RemoteSubscription) {
-	s.remoteSubsLock.Lock()
-	defer s.remoteSubsLock.Unlock()
-
-	s.remoteSubs[topic] = remoteSub
+func (s *Session) addRemoteSession(sid string, rs *remoteSession) {
+	s.remoteSessionsLock.Lock()
+	defer s.remoteSessionsLock.Unlock()
```

Comment:
`defer` is useful when you have multiple returns from the function if you need to return a value from a guarded operation. In simple cases like this it's a better practice to call `Unlock` directly because it's more efficient and uses less memory.

Reply:
Done.